internal/bminventory/inventory.go

```diff
@@ -1809,17 +1816,13 @@ func (b *bareMetalInventory) updateNonDhcpNetworkParams(updates map[string]inter
 				return common.NewApiError(http.StatusBadRequest, errors.Wrap(err, "Calculate machine network CIDR"))
 			}
 			if primaryMachineNetworkCidr != "" {
-				if network.IsMachineCidrAvailable(cluster) {
-					cluster.MachineNetworks[0].Cidr = models.Subnet(primaryMachineNetworkCidr)
+				// We set the machine networks in the ClusterUpdateParams, so they will be viewed as part of the request
+				// to update the cluster
+				if len(params.ClusterUpdateParams.MachineNetworks) > 0 {
+					params.ClusterUpdateParams.MachineNetworks[0].Cidr = models.Subnet(primaryMachineNetworkCidr)
 				} else {
-					cluster.MachineNetworks = []*models.MachineNetwork{{Cidr: models.Subnet(primaryMachineNetworkCidr)}}
+					params.ClusterUpdateParams.MachineNetworks = []*models.MachineNetwork{{Cidr: models.Subnet(primaryMachineNetworkCidr)}}
 				}
-				// In case we calculated Machine Network automatically (because conditions for
-				// doing so were met), we update the `params` payload so that from the
-				// perspective of the DB write/delete actions it looks like a valid update
-				// operation. Without that we don't have other way to mark this field of the
-				// cluster as dirty.
-				params.ClusterUpdateParams.MachineNetworks = []*models.MachineNetwork{{Cidr: models.Subnet(primaryMachineNetworkCidr)}}
 			}
 		}
```

Comment:
I'm just thinking, in this part of code... Previously we were updating both `cluster.MachineNetworks` and `params.ClusterUpdateParams.MachineNetworks`. With this change we will be updating only `params.ClusterUpdateParams.MachineNetworks`. Is it clear who will take care of updating `cluster.MachineNetworks` and when?

I recall from the past that there were some problems with the value being updated, but at a wrong time. So random places in the code were referring to the value post-update and some to the value pre-update. Just trying to understand how is this move going to affect the flow.

I trust that the change here is not breaking stuff, but can't prove it by just looking at this piece

Reply:
I tried to make it more consistent.  ClusterUpdateParams is what we want to update, and the cluster itself is the current status.
The networks are updated in this function https://github.com/openshift/assisted-service/blob/1df12c082e044f862dcfe847d028918b76a16393/internal/bminventory/inventory.go#L2056

And are verified here

https://github.com/openshift/assisted-service/blob/1df12c082e044f862dcfe847d028918b76a16393/internal/bminventory/inventory.go#L1983